src/libpriv/rpmostree-postprocess.cxx

```diff
@@ -945,3 +945,26 @@ rpmostree_compose_commit (int            rootfs_fd,
 
   return TRUE;
 }
+
+namespace rpmostreecxx {
+rust::String
+compose_commit (int            rootfs_fd,
+                OstreeRepo    *repo,
+                const char    *parent_revision,
+                GVariant      *src_metadata,
+                const char    *gpg_keyid,
+                bool           enable_selinux,
+                OstreeRepoDevInoCache *devino_cache,
+                GCancellable  *cancellable)
+{
+  g_autoptr(GError) local_error = NULL;
+  g_autofree char *new_revision = NULL;
+  try {
+    if (!internal_rpmostree_compose_commit (rootfs_fd, repo, parent_revision, src_metadata, gpg_keyid, enable_selinux, devino_cache, &new_revision, cancellable, &local_error))
+      util::throw_gerror(local_error);
+  } catch (std::exception &e) {
+    util::rethrow_prefixed(e, "Writing commit");
+  }
+  return rust::String(new_revision);
```

Comment:
Note this tricky dance is required to add a prefix in both cases:

1) Underlying code directly throws a C++ exception
2) GError is set